File: meeting-scheduler/meeting-scheduler.cs
public class Solution {
    public IList<int> MinAvailableDuration(int[][] slots1, int[][] slots2, int duration) {
        Array.Sort(slots1, (x, y) => {return x[0] - y[0];});
        Array.Sort(slots2, (x, y) => {return x[0] - y[0];});

        int p1 = 0;
        int p2 = 0;
        while(p1 < slots1.Length && p2 < slots2.Length) {
            int left = Math.Max(slots1[p1][0], slots2[p2][0]);
            int right = Math.Min(slots1[p1][1], slots2[p2][1]);
            if(right - left >= duration) {
                return new List<int>(){left, left + duration};
            }
            if(slots1[p1][1] < slots2[p2][1]) {
                p1++;
            }
            else {
                p2++;
            }
        }
        return new List<int>();
    }
}

File: meeting-scheduler/meeting-scheduler.java
class Solution {
    public List<Integer> minAvailableDuration(int[][] slots1, int[][] slots2, int duration) {
        int i = 0, j = 0;
        Arrays.sort(slots1, (a, b) -> a[0] - b[0]);
        Arrays.sort(slots2, (a, b) -> a[0] - b[0]);
        while(i < slots1.length && j < slots2.length) {
            int[] slot1 = slots1[i];
            int[] slot2 = slots2[j];
            int start = Math.max(slot1[0], slot2[0]);
            int end = Math.min(slot1[1], slot2[1]);
            if(end - start >= duration) {
                return Arrays.asList(start, start + duration);
            }
            if(slot1[1] == end) {
                i++;
            }
            if(slot2[1] == end) {
                j++;
            }
        }
        return new ArrayList<>();
    }
}